ASPNetCoreMasters/Controllers/ItemsController.cs

```diff
@@ -10,32 +12,74 @@
 
 namespace ASPNetCoreMasters.Controllers
 {
+    [ApiController]
+    [Route("[controller]")]
     public class ItemsController : ControllerBase
-    {   
-        public int Get(int id)
-        {
-            //var item = new ItemService();
-            //var result = item.GetAll(id);
+    {
+        private readonly IItemService _itemService;
 
-            return id;
+        public ItemsController(IItemService itemService)
+        {
+            _itemService = itemService;
         }
 
-        public IActionResult Post([FromBody] ItemCreateBindingModel itemCreateBidingModel)
+        [HttpGet]
+        public IActionResult GetAll()
         {
-            var item = new ItemService();
-            var itemDTO = new ItemDTO();
+            var result = _itemService.GetAll();
+            return Ok(result);
+        }
 
-            if (!ModelState.IsValid)
+        [HttpGet("{itemId}")]
+        public IActionResult Get(int itemId)
+        {
+            if (String.IsNullOrEmpty(itemId.ToString()))
```

Comment:
if itemId is null or missing then the GetAll action will be hit. If it doesn't match any route then by default a 404 not found will be returned so there is no need for this checking. No need to remove for approval, just for your info